### PR TITLE
Joss paper

### PR DESCRIPTION
**Pull request by rogerkuou**
Add a JOSS paper for MOTrainer.

### CODE REVIEW

**Comment by rogerkuou** [comment]
Hi all, I have drafted a JOSS paper for MOTrainer. You have all contributed to this repository therefore I listed you all as authors. Would you like to look at this paper? 

It should be rendered as a GitHub Action artifact. You can download it (need to be unzipped) from [here](https://github.com/VegeWaterDynamics/motrainer/actions/runs/7473887442/artifacts/1159197901).

Thanks in advance!

File: paper/paper.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+   index: 1
+ - name: Delft University of Technology, Netherlands
+   index: 2
+date: 22 Dec 2023
+bibliography: paper.bib
+---
+
+## Summary
+
+Data Assimilation (DA) remains a pivotal data analytical technique in environmental science research, enabling the constraint of physical model states with observation data.
```
Comment (rogerkuou) [inline paper/paper.md:37]:
```suggestion
Data assimilation is an essential tool in Earth and environmental sciences, enabling physical model states to be constrained using observational data
```

File: paper/paper.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+In the DA process, observation data is integrated into the physical model through the application of a Measurement Operator (MO) – a connection model mapping physical model states to observations. Researchers have observed that employing a Machine-Learning (ML) model as a surrogate MO can bypass the limitations associated with using an overly simplified MO [@Forman:2014; @XUE:2015; @Forman:2017].
```
Comment (rogerkuou) [inline paper/paper.md:39]:
```suggestion
In the DA process, observations are integrated into the physical model through the application of a Measurement Operator (MO) – a connection model mapping physical model states to observations. Researchers have observed that employing a Machine-Learning (ML) model as a surrogate MO can bypass the limitations associated with using an overly simplified MO [@Forman:2014; @XUE:2015; @Forman:2017].
```

File: paper/paper.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+## Statement of Need
+
+ML models should undergo training within a coherent spatio-temporal scope, where physical model states can be consistently mapped to observations using the same model. Dealing with a large spatio-temporal scale may involve multiple mapping processes, necessitating the consideration of training separate models for distinct spatial and/or temporal partitions of the dataset. As the number of partitions increases, a challenge emerges in effectively distributing these training tasks among the partitions.
+
+A surrogate MO, as a ML model, should be trained over a coherent spatio-temporal scope, where one can assume that the same MO applies when mapping physical model states to observations. When dealing with a large spatio-temporal scale, multiple mapping processes may exist, prompting consideration for training separate MOs for distinct spatial and/or temporal partitions of the dataset. As the number of partitions increases, a challenge arises in distributing these training tasks effectively among the partitions.
```
Comment (rogerkuou) [inline paper/paper.md:45]:
```suggestion
A surrogate MO, as a ML model, should be trained with the assumption that a single MO applies when mapping physical model states to observations. When dealing with a large spatio-temporal scale, multiple mapping processes may exist, prompting consideration for training separate MOs for distinct spatial and/or temporal partitions of the dataset. As the number of partitions increases, a challenge arises in distributing these training tasks effectively among the partitions.
```

**Review by rogerkuou — COMMENTED** [review]
Applied some comments made by Susan

File: paper/paper.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+## Summary
+
+Data Assimilation (DA) remains a pivotal data analytical technique in environmental science research, enabling the constraint of physical model states with observation data.
+
+In the DA process, observation data is integrated into the physical model through the application of a Measurement Operator (MO) – a connection model mapping physical model states to observations. Researchers have observed that employing a Machine-Learning (ML) model as a surrogate MO can bypass the limitations associated with using an overly simplified MO [@Forman:2014; @XUE:2015; @Forman:2017].
+
+## Statement of Need
+
+ML models should undergo training within a coherent spatio-temporal scope, where physical model states can be consistently mapped to observations using the same model. Dealing with a large spatio-temporal scale may involve multiple mapping processes, necessitating the consideration of training separate models for distinct spatial and/or temporal partitions of the dataset. As the number of partitions increases, a challenge emerges in effectively distributing these training tasks among the partitions.
```
Comment (rogerkuou) [inline paper/paper.md:43]:
```suggestion
```

File: paper/paper.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+To address this challenge, we developed the open Python package `MOTrainer`. It caters to researchers requiring training independent MOs across extensive spatio-temporal coverage in a distributed manner. `MOTrainer` leverages Xarray's support for multi-dimensional datasets to accommodate spatio-temporal features of input/output data of the training tasks. It provides user-friendly functionalities implemented with the Dask library, facilitating the partitioning of large spatio-temporal data for independent model training tasks. Additionally, it streamlines the train-test data split based on customized spatio-temporal coordinates. The Jackknife method [@mccuen1998hydrologic] is implemented as an external Cross-Validation (CV) method for Deep Neural Network (DNN) training, with support for Dask parallelization. This feature enables the scaling of training tasks across various computational infrastructures.
```
Comment (rogerkuou) [inline paper/paper.md:46]:
```suggestion
To address this challenge, we developed a novel approach for distributed training of MOs. We present the open Python library `MOTrainer`, which to the best of our knowledge, is the first Python library catering to researchers requiring training independent MOs across extensive spatio-temporal coverage in a distributed manner. `MOTrainer` leverages Xarray's support for multi-dimensional datasets to accommodate spatio-temporal features of input/output data of the training tasks. It provides user-friendly functionalities implemented with the Dask library, facilitating the partitioning of large spatio-temporal data for independent model training tasks. Additionally, it streamlines the train-test data split based on customized spatio-temporal coordinates. The Jackknife method [@mccuen1998hydrologic] is implemented as an external Cross-Validation (CV) method for Deep Neural Network (DNN) training, with support for Dask parallelization. This feature enables the scaling of training tasks across various computational infrastructures.
```

File: paper/paper.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+   index: 1
+ - name: Delft University of Technology, Netherlands
+   index: 2
+date: 22 Dec 2023
+bibliography: paper.bib
+---
+
+## Summary
+
+Data assimilation is an essential tool in Earth and environmental sciences, enabling physical model states to be constrained using observational data.
```
Comment (SarahAlidoost) [inline paper/paper.md:37]:
```suggestion
Data assimilation (DA) is an essential procedure in Earth and environmental sciences, enabling physical model states to be constrained using observational data.
```

File: paper/paper.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+In the DA process, observations are integrated into the physical model through the application of a Measurement Operator (MO) – a connection model mapping physical model states to observations. Researchers have observed that employing a Machine-Learning (ML) model as a surrogate MO can bypass the limitations associated with using an overly simplified MO [@Forman:2014; @XUE:2015; @Forman:2017].
+
+## Statement of Need
+
+A surrogate MO, as a ML model, should be trained with the assumption that a single MO applies when mapping physical model states to observations. When dealing with a large spatio-temporal scale, multiple mapping processes may exist, prompting consideration for training separate MOs for distinct spatial and/or temporal partitions of the dataset. As the number of partitions increases, a challenge arises in distributing these training tasks effectively among the partitions.
```
Comment (SarahAlidoost) [inline paper/paper.md:43]:
```suggestion
A surrogate MO, as a ML model, is trained with the assumption that a single MO applies when mapping physical model states to observations. When dealing with a large spatio-temporal scale, multiple mapping processes may exist, prompting consideration for training separate MOs for distinct spatial and/or temporal partitions of the dataset. As the number of partitions increases, a challenge arises in distributing these training tasks effectively among the partitions.
```

**Review by SarahAlidoost — APPROVED** [review]
@rogerkuou thank you :+1: . Just some minor comments.

File: paper/paper.md (hunk 7 above)
Comment (rogerkuou) [reply paper/paper.md:43]: @rogerkuou thanks for the suggestion! Sorry I updated you change by further deleting that comma.

File: paper/paper.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+   index: 2
+date: 22 Dec 2023
+bibliography: paper.bib
+---
+
+## Summary
+
+Data assimilation is an essential tool in Earth and environmental sciences, enabling physical model states to be constrained using observational data.
+
+In the DA process, observations are integrated into the physical model through the application of a Measurement Operator (MO) – a connection model mapping physical model states to observations. Researchers have observed that employing a Machine-Learning (ML) model as a surrogate MO can bypass the limitations associated with using an overly simplified MO [@Forman:2014; @XUE:2015; @Forman:2017].
```
[the remark this answers is not in the document]
Comment (rogerkuou) [reply paper/paper.md:39]: I wanted to make a start to talk about Measurement Operators. So I stared with the function of MO in a DA process. Maybe I should say `In a DA process, ....`?

**Pull request merged**